homework-2/script/maim.js

```diff
@@ -0,0 +1,318 @@
+        const create = await fetch( url, {
+            method: 'POST',
+            body: JSON.stringify(obj),
+              headers: {
+                'Content-type': 'application/json; charset=UTF-8',
+              },
+        });
+        
+        if (!create.ok) {
+            throw new Error(`Someting went wrong, Status code ${create.status}`)
+        }
+        return await create.json();
+    }
+
+
+    
+    const deleteUser = async (id) => {
+        const response = await fetch(`${url}/${id}`, {
+            method: 'DELETE'
+        });
+
+        if (!response.ok) {
+            throw new Error(`Someting went wrong, Status code ${response.status}`)
+        }
+        return await response.json();
+    }
+
+
+
+    const updateUser = async (id, obj) => {
+        const update = await fetch(`${url}/${id}`, {
+            method: 'PATCH',
+            body: JSON.stringify(obj),
+              headers: {
+                'Content-type': 'application/json; charset=UTF-8',
+              },
+        });
+        
+        if (!update.ok) {
+            throw new Error(`Someting went wrong, Status code ${update.status}`)
+        }
+        return await update.json();
+    }
+
+
+
+
+    getAllUsers(url)
+        .then(value => value.reverse())
+        .then(data => {
+            paginationUser(data); 
+        });
+
+
+    filterUserBtn.addEventListener('click', () => {
+        getAllUsers(url)
+        .then(data => {
+                let filterDataCompany = data.filter(value => value.company.startsWith( userFilterCompany.value.toUpperCase() ));
+                let filterDataName = filterDataCompany.filter(value => value.name.startsWith(userFilterName.value.toUpperCase()));
+                let filterDataAddress = filterDataName.filter(value => value.address.startsWith(userFilterAddress.value.toUpperCase()));
+
+                renderUserInfo(filterDataAddress);
+
+                paginationPage.style.display = 'none';
+
+                if (!userFilterCompany.value && !userFilterName.value && !userFilterAddress.value) {
+                    showPage(navigationBtn[0], data);
+                    paginationPage.style.display = 'flex';
+                }
+        });
+    });
+
+
+    let notesOnPage = 5;
+
+    function paginationUser(data) {
+        let counOfPages = Math.ceil(data.length / notesOnPage);
+        
+
+        for (let i = 1; i <= counOfPages; i++) {
+        let paginationBtn = document.createElement('li');
+            paginationBtn.classList.add('pagination-list');
+            navigationBtn.push( paginationBtn );
+            paginationBtn.textContent = i;
+            paginationPage.append( paginationBtn );
+        }
+
+        showPage(navigationBtn[0], data);
+        
+        navigationBtn.forEach((btn) => {
+            btn.addEventListener('click', function() {
+                showPage( this, data );
+            });
+        });
+    }
+
+    let activeList;
+    function showPage(item, data) {
```

Comment:
you have to re-initiate pagination after filtering, it now shows the results on the current page and should only display the filtered results and show how many pages of filtered items there are

Reply:
![image](https://user-images.githubusercontent.com/84564899/127327983-cfd33c04-fef8-4b74-a1bd-525edf75062e.png)